### PR TITLE
feat: implement a goclient example

### PR DESCRIPTION
**Pull request by D4ryl00**
This is an implementation of a Go client/server of Gnomobile.

- [x] add an error type when the init of the service failed
- [x] modify Config to use UDS by default. Add the ability to change UDS/TCP addresses.
- [x] implementation of a Go server of Gnomobile.
- [x] implementation of a Go client of Gnomobile (UDS are not implemented yet).

### CODE REVIEW

File: goserver/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Go server
+
+This Go executable run a Gnomobile service.## Run the service
+
+`go run ./goserver udp`
```
Comment (jefft0) [inline goserver/README.md:5]: Maybe "udp" should be "tcp"

File: goserver/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+or
+
+`go run ./goserver udp`
```
Comment (jefft0) [inline goserver/README.md:9]: This is a repeated. Maybe this is the option to specify the tcp port?

**Review by jefft0 — APPROVED** [review]
See my minor doc comments. You can fix and merge. I tested the goclient. LGTM.

File: goserver/README.md (hunk 1 above)
Comment (D4ryl00) [reply goserver/README.md]: thx, it was `uds` or `tcp`

**Pull request merged**